base/meta.jl

```diff
@@ -363,10 +363,15 @@ function _partially_inline!(@nospecialize(x), slot_replacements::Vector{Any},
         return x
     end
     if isa(x, Core.ReturnNode)
+       # Unreachable doesn't have val defined
+       if !isdefined(x, :val)
+          return x
+       else
         return Core.ReturnNode(
             _partially_inline!(x.val, slot_replacements, type_signature, static_param_values,
                                slot_offset, statement_offset, boundscheck),
         )
+       end
     end
     if isa(x, Core.GotoIfNot)
         return Core.GotoIfNot(
```

Comment:
reasonable, though unfortunately knowing how to handle that is a bit out of my depth =/. I found this while working on https://github.com/EnzymeAD/Reactant.jl/pull/364 (which if you have availability, I'd love comment/review on ... especially since apparently I segfault julia). Where...we do use a variant of this function for post inference IR (and I'm not sure what else to use).